security/pkg/server/ca/server.go

```diff
@@ -153,29 +153,20 @@ func recordCertsExpiry(keyCertBundle *util.KeyCertBundle) {
 	if err != nil {
 		serverCaLog.Errorf("failed to extract root cert expiry timestamp (error %v)", err)
 	}
-	rootCertExpiryTimestamp.Record(rootCertExpiry)
-
-	rootCertPem, err := util.ParsePemEncodedCertificate(keyCertBundle.GetRootCertPem())
-	if err != nil {
-		serverCaLog.Errorf("failed to parse the root cert: %v", err)
-	}
-	rootCertExpirySeconds.ValueFrom(func() float64 { return time.Until(rootCertPem.NotAfter).Seconds() })
+	rootCertExpiryTimestamp.Record(float64(rootCertExpiry.Unix()))
+	rootCertExpirySeconds.ValueFrom(rootCertExpiry.Sub(time.Now()).Seconds)
```

Comment:
Doing this is also a bit nicer because we only need to parse the cert bundle once, which saves us a mutex lock, copy, and parse.